### PR TITLE
Updated install scripts and removed frontend build from install

### LINKED ISSUE #522: Remove frontend UT suite execution from setup
## Is your feature request related to a problem? Please describe.
Dusing install from source, Gradle build executes complete frontend UT suite thereby increasing the install time by a considerable amount.

## Describe the solution
Setup should only `generate` the required frontend jar for installation and not execute the whole frontend UT suite. 

Developers have the option to run frontend UT suite manually and this is also included in the torchserve_sanity script which is a part of the CI build process.

### PR DESCRIPTION
**Pull request by harshbafna**
## Description

 - replaced complete frontend build with `assemble` in `setup.py`, thereby removing the un-necessary UT suite execution on install from source.
 - unified install_from_src scripts for macOS and ubuntu
 - updated documentation for following
  - Code coverage and sanity script execution
  - Contributing readme
  - Main README as per changes in above scripts

Fixes #522

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Feature/Issue validation/testing

[install from source mac.txt](https://github.com/pytorch/serve/files/4912878/install.from.source.mac.txt)
[install from source ubuntu.txt](https://github.com/pytorch/serve/files/4912880/install.from.source.ubuntu.txt)
[install from src ubuntu gpu.txt](https://github.com/pytorch/serve/files/4912882/install.from.src.ubuntu.gpu.txt)
[torchserve sanity macos.txt](https://github.com/pytorch/serve/files/4912884/torchserve.sanity.macos.txt)
[torchserve sanity ubuntu.txt](https://github.com/pytorch/serve/files/4912885/torchserve.sanity.ubuntu.txt)
[torchserve sanity ubuntu gpu.txt](https://github.com/pytorch/serve/files/4912889/torchserve.sanity.ubuntu.gpu.txt)

## Checklist:

- [x] Have you added tests that prove your fix is effective or that this feature works?
- [x] New and existing unit tests pass locally with these changes?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [x] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -7,7 +7,12 @@ If you are interested in contributing to TorchServe, your contributions will fal
 2. You want to implement a feature or bug-fix for an outstanding issue.
     - Search for your issue here: https://github.com/pytorch/serve/issues
     - Pick an issue and comment on the task that you want to work on this feature.
-   - If you need more context on a particular issue, please ask.
+    - To ensure your changes doesn't break any of the existing features run the sanity suite as follows from serve directory:
+    ```bash
+    ./torchserve_sanity.sh
+    ```
+    - For running individual test suites refer [code_coverage](docs/code_coverage.md) documentation
+    - If you need more context on a particular issue, please ask.
```
Comment (maaquib) [inline CONTRIBUTING.md:15]: Can we change this to the slack channel reference and other means of communication. **please ask** is vague

File: docs/code_coverage.md [hunk 2]
```diff
@@ -1,30 +1,55 @@
...
+```bash
+python -m pytest --cov-report html:htmlcov --cov=ts/ ts/tests/unit_tests/
+```
+
+The reports are generated at following path : `htmlcov/`
+
+## To run python linting on `ts` package run following command
+
+```bash
+pylint -rn --rcfile=./ts/tests/pylintrc ts/.;
```
Comment (maaquib) [inline docs/code_coverage.md:33]:
```suggestion
pylint -rn --rcfile=./ts/tests/pylintrc ts/.
```

File: setup.py [hunk 3]
```diff
@@ -77,12 +78,8 @@ def run(self):
         if os.path.exists(self.source_server_file):
             os.remove(self.source_server_file)
 
-        # Remove build/lib directory.
-        if os.path.exists('build/lib/'):
```
Comment (maaquib) [inline setup.py:81]: Whats the reason for removing this?

Comment (harshbafna) [reply setup.py:81]: When we use `assemble` the build directory is never generated.

**Comment by harshbafna** [comment]
@chauhang 

> 1. Prev version of the script used to check if launching from conda env or outside. Can you please verify that check is not needed anymore

Earlier there was just an echo statement asking the user to "exit if using a conda enviornment". However, we have been using the install from source scripts on DL AMIs and mac with conda since the first release without any issues. Hence removed the statement.

> 2. How to do install from src on GPU machines? Can you please update the scripts and readme for this?

The same script works on both CPU and GPU.

File: CONTRIBUTING.md (hunk 1 above)
Comment (harshbafna) [reply CONTRIBUTING.md]: Done.

**Comment by dhanainme** [comment]
@chauhang @maaquib , This looks good to me. 

Will go ahead & merge this today if you dont have anymore feedback

**Comment by maaquib** [comment]
Verified on GPU and CPU
```
./scripts/install_from_src
torchserve --start --model-store /tmp/model_store/
curl -X POST "http://localhost:8081/models?url=https://torchserve.s3.amazonaws.com/mar_files/my_text_classifier.mar&initial_workers=1&synchronous=true"
```

**Pull request merged**